app/main/posts/views/post-view-data.directive.js

```diff
@@ -154,6 +164,11 @@ function PostViewDataController(
         checkForNewPosts(30000);
     }
 
+    function persistUpdatedPost(updatedPost) {
+        _.extend(_.findWhere($scope.posts, { id: updatedPost.id }), updatedPost);
```

Comment:
The use of extend could get a bit weird here if you remove a value from the post, it might not get removed.
It might be better to use `splice()` instead to swap the entire posts object out of `$scope.posts`

Reply:
Sure